lib/sidekiq/job_util.rb

```diff
@@ -94,16 +94,16 @@ def json_safe?(item)
     end
 
     def json_unsafe_item(item)
-      case item
-      when String, Integer, Float, TrueClass, FalseClass, NilClass
+      klass = item.class
+      if [String, Integer, Float, TrueClass, FalseClass, NilClass].include?(klass)
         nil
-      when Array
+      elsif klass == Array
         item.each do |e|
           unsafe_item = json_unsafe_item(e)
           return unsafe_item unless unsafe_item.nil?
         end
         nil
-      when Hash
+      elsif klass == Hash
```

Comment:
Should we allow Hash and Array subclasses with the assumption they have the same behavior for serialization purposes? I think HWIA is going to be a big support issue otherwise.

Reply:
These classes are already rejected by `json_safe?`. Should we enable it there too?

If we have HWIA with symbol keys, they will be serialized as string keys (and not a HWIA in the `perform`, as the user may be expecting). The user will try to access them via a symbol key and get a nil. If we assume, that the user knows and remembers that hashes in `perform` always have string keys, then probably yes. Wdyt?

Reply:
I believe HWIA stores keys internally as Strings so we're safe in that regard.

So we either fail in serializing or it will silently fail when they use Symbols to access hash values. Ugh. We have to fail early and loudly. Failing silently allows bugs to slip past. You're right, we can't allow HWIA.